sklearn/datasets/openml.py

```diff
@@ -250,7 +250,12 @@ def _get_data_info_by_name(name, version, data_home):
         error_msg = "No active dataset {} found.".format(name)
         json_data = _get_json_content_from_openml_api(url, error_msg, True,
                                                       data_home)
-        return json_data['data']['dataset'][0]
+        res = json_data['data']['dataset']
+        if len(res) > 1:
+            warn("Multiple active versions of the dataset matching the name"
+                 " {name} exist. Versions may be different, returning version"
```

Comment:
I think the word "very"  or "fundamentally" is important here. Of course versions may be different